### DISCUSSION TITLE
primarySite Twig global variable

### DISCUSSION THREAD

**Discussion by jamesmacwhite** [craftcms/cms#16370] [Ideas] [answered]
Similar to the currentSite variable, but returning the primarySite model instead.

My reason for suggesting, if I wanted to use the primary site properties e.g. base URL in another site, I currently would need to do a query with craft.app.sites, or use the siteUrl function but the params are little awkward e.g. to get the homepage of the primary site `siteUrl('', null, null, 1)`, with 1 being the ID of primary site.

A syntax of:

```
{{ primarySite.name }}
{{ primarySite.baseUrl }}
```

Would be cleaner.

**Comment by mmikkel** [comment]
You can get the primary site via `craft.app.sites.primarySite`, but I'd still +1 this – needing to refer to the primary site in multi-site builds is pretty common, so having a global `primarySite` variable/shortcut would be a nice convenience feature.

@jamesmacwhite That `siteUrl()` example is a perfect use case for [named arguments](https://twig.symfony.com/doc/3.x/templates.html#named-arguments) in Twig 😉:

```twig
{{ siteUrl(siteId=1) }}
```

Or, if you want to avoid hard coding the ID:

```twig
{{ siteUrl(siteId=craft.app.sites.primarySite.id) }}
```

It would be cool to be able to do things like this though:

```twig
{{ siteUrl('login', siteId=primarySite.id) }}
```

**Comment by jamesmacwhite** [comment]
@mmikkel Thanks! I've learnt something new with named arguments today with Twig, to make siteUrl usage a bit cleaner for sure!

**Comment by brandonkelly** [comment, marked as answer]
Added for Craft 4.14 and 5.6! (92cdb09fe32686eb1cf034ce495b2a54149a3db3)

**Reply by jamesmacwhite** [reply]
Short syntax rejoice! Thanks Brandon!

**Reply by brandonkelly** [reply]
Craft 4.14.0 and 5.6.0 are out now! Thanks for the suggestion @jamesmacwhite.
